lib/streamlit/watcher/event_based_path_watcher.py

```diff
@@ -340,6 +343,7 @@ def handle_path_change_event(self, event: events.FileSystemEvent) -> None:
             return
 
         changed_path = os.path.abspath(changed_path)
+        assert isinstance(changed_path, str)
```

Comment:
What was the underlying type error here? Isn't `abspath` gurunteed to return a str? Would it also solve it if we write this to a new variable, e.g. `changed_abs_path`?

Reply:
I was wrong before. Apparantly `os.path.abspath` can return a bytes object if you pass it a bytes object.

The core of the issue was that `event.src_path` and `event.dest_path` can now technically return both a string and byte object. To fix this I added a decoder that decodes the bytes ojbect to a string in the case it is a bytes object.

Reply:
Apologies for the many pushes to this PR!

Somehow a the Mypy check for Python min (3.8?) still failed. I think I found a solution now, so hopefully this fill fix the last Mypy error. :)